sdk/core/core-lro/src/poller.ts

```diff
@@ -0,0 +1,133 @@
+import { EventEmitter } from "events";
+import { delay, HttpOperationResponse, RequestOptionsBase, ServiceClient } from "@azure/core-http";
+
+export interface PollerOptionalParameters<States> {
+  manual?: boolean;
+  intervalInMs?: number;
+  state?: States;
+  client?: ServiceClient;
+  initialResponse?: HttpOperationResponse;
+  previousResponse?: HttpOperationResponse;
+  requestOptions?: RequestOptionsBase;
+  stopped?: boolean;
+}
+
+export abstract class Poller<States, Result> extends EventEmitter  {
+  private readonly manual: boolean = false;
+  private stopped: boolean = false;
+  private intervalInMs: number = 1000;
+  private _state: States;
+  protected readonly client: ServiceClient;
+  protected initialResponse?: HttpOperationResponse;
+  protected previousResponse?: HttpOperationResponse;
+  protected requestOptions: RequestOptionsBase | undefined;
+  public stopped: boolean;
+
+  constructor(options: PollerOptionalParameters) {
+    this.manual = Boolean(options.manual);
+    this.intervalInMs = Number(options.intervalInMs);
+    this.client = options.client;
+    this.initialResponse = options.initialResponse;
+    this.previousResponse = options.previousResponse;
+    this.requestOptions = options.requestOptions;
+    this.stopped = Boolean(options.stopped);
+
+    if (options.state) this._state = options.state;
+    if (this.manual) return;
+    this.poll();
+  }
+
+  protected abstract getStateFromResponse(
+    response: HttpOperationResponse
+  ): LongRunningOperationStates;
+
+  protected abstract async sendRequest(options?: RequestOptionsBase): Promise<HttpOperationResponse>;
+
+  protected async processResponse(response: HttpOperationResponse): Promise<void | Result> {
+    if (!this.initialResponse) this.initialResponse = response;
+    this.previousResponse = response;
+    this.emit("newResponse", response, this);
```

Comment:
shouldn't this be fired **after** state is updated? Otherwise the client will not be able to take advantage of any state updating the subclass has performed